build/webTestReporter.js

```
@@ -73,24 +78,131 @@ exports.startReportServer = async function () {
+                                case 'error':
+                                    return `${colorizedTime}${colors.red(item.output)}`;
+                                default:
+                                    return `${colorizedTime}${item.output}`;
+                                    break;
+                            }
+                        })
+                        .map((item) => `${item}\n`);
+                    const consoleOutput = {
+                        name: 'stdout',
+                        output_type: 'stream',
+                        text: consoleOutputs
+                    };
+                    // Look for a screenshot file with the above prefix & attach that to the cell outputs.
+                    const screenshots = glob
+                        .sync(`${fileNamePrefix}*-screenshot.png`, { cwd: ExtensionRootDir })
+                        .map((file) => {
+                            const contents = Buffer.from(fs.readFileSync(path.join(ExtensionRootDir, file))).toString(
+                                'base64'
+                            );
+                            return {
+                                data: {
+                                    'image/png': contents
+                                },
+                                metadata: {},
+                                output_type: 'display_data'
+                            };
+                        });
+                    cells.push({
+                        cell_type: 'code',
+                        metadata: {
+                            collapsed: true
+                        },
+                        source: `#${output.title}`,
+                        execution_count: ++executionCount,
+                        outputs: [assertionError, consoleOutput, ...screenshots]
+                    });
+                    break;
+                }
+            }
```

Could it be possible to have a case that doesn't fit any of the other cases? I'm thinking we could add a simple "default" case to throw (in case of an unexpected behavior or condition).

---

No, for other cases we don't need to do anything, hence throwing an exception is wrong
basically this switch is not handling all possible cases, but only handling what we need.

---

oooh perfect! thank you 👍